app/controllers/storage_locations_controller.rb

```diff
@@ -18,9 +18,16 @@ def index
       @storage_locations = @storage_locations.kept
     end
 
+    inventory_item_names = []
+    InventoryItem.all.each{ |inv_item|
+      cur_item = Item.find(inv_item.item_id)
+      inventory_item_names << cur_item.name if cur_item.active && !inventory_item_names.include?(cur_item.name)
+    }
+    inventory_item_names.sort!{ |a, b| a <=> b }
+
```

Comment:
In general we want to do as much processing in the database as possible, so you can make use of indexes. If you're trying to get all the item names belonging to the storage location, you can do it thusly:

```ruby
@storage_location.
  inventory_items. # get all inventory items in the storage location
  includes(:item). # get the corresponding item for each inventory item
  joins(:item). # include the item in the join query so we can filter it
  select('distinct items.name'). # only get the name out of the items table
  where(:items => { active: true}). # only get items that are active
  map(&:name). # get the name from each result
  sort # sort alphabetically
```